### PR TITLE
Changed the BYOK module to create an IAM role and only one CMK

### PR DESCRIPTION
**Pull request by deepak-clumio**

### CODE REVIEW

**Comment by deepak-clumio** [comment]
```
$ AWS_PROFILE=customer terraform apply -var-file=terraform.tfvar
module.clumio_byok.random_uuid.external_id: Refreshing state... [id=ea74ed66-95ea-492b-228d-638cd4052216]
clumio_wallet.test_wallet: Refreshing state... [id=455372684561]
module.clumio_byok.aws_kms_key.multi_region_cmk_key[0]: Refreshing state... [id=mrk-c31e2af81d8c4f46a8989bd4c768afeb]

Note: Objects have changed outside of Terraform

Terraform detected the following changes made outside of Terraform since the last "terraform apply":

  # module.clumio_byok.aws_kms_key.multi_region_cmk_key[0] has changed
  ~ resource "aws_kms_key" "multi_region_cmk_key" {
        id                                 = "mrk-c31e2af81d8c4f46a8989bd4c768afeb"
      + tags                               = {}
        # (12 unchanged attributes hidden)
    }


Unless you have made equivalent changes to your configuration, or ignored the relevant attributes using ignore_changes, the following plan may include actions to undo or respond to these changes.

────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # module.clumio_byok.aws_iam_role.byok_mgmt_role will be created
  + resource "aws_iam_role" "byok_mgmt_role" {
      + arn                   = (known after apply)
      + assume_role_policy    = jsonencode(
            {
              + Statement = [
                  + {
                      + Action    = "sts:AssumeRole"
                      + Condition = {
                          + "ForAnyValue:StringEquals" = {
                              + "sts:ExernalId" = "ea74ed66-95ea-492b-228d-638cd4052216"
                            }
                        }
                      + Effect    = "Allow"
                      + Principal = {
                          + AWS = "arn:aws:iam::972749477403:root"
                        }
                      + Sid       = "AllowClumioToAssumeRole"
                    },
                ]
              + Version   = "2012-10-17"
            }
        )
      + create_date           = (known after apply)
      + force_detach_policies = false
      + id                    = (known after apply)
      + managed_policy_arns   = (known after apply)
      + max_session_duration  = 3600
      + name                  = "ClumioKMSRole-tf-Org7"
      + name_prefix           = (known after apply)
      + path                  = "/"
      + tags_all              = (known after apply)
      + unique_id             = (known after apply)

      + inline_policy {
          + name   = "ReplicaAndGrantManagement"
          + policy = jsonencode(
                {
                  + Statement = [
                      + {
                          + Action   = [
                              + "kms:TagResource",
                              + "kms:RevokeGrant",
                              + "kms:RetireGrant",
                              + "kms:ReplicateKey",
                              + "kms:PutKeyPolicy",
                              + "kms:ListRetirableGrants",
                              + "kms:ListGrants",
                              + "kms:DescribeKey",
                              + "kms:CreateGrant",
                            ]
                          + Effect   = "Allow"
                          + Resource = "arn:aws:kms:*:455372684561:key/mrk-c31e2af81d8c4f46a8989bd4c768afeb"
                          + Sid      = ""
                        },
                      + {
                          + Action   = "kms:CreateKey"
                          + Effect   = "Allow"
                          + Resource = "*"
                          + Sid      = ""
                        },
                    ]
                  + Version   = "2012-10-17"
                }
            )
        }
    }

  # module.clumio_byok.clumio_post_process_kms.clumio_phone_home will be created
  + resource "clumio_post_process_kms" "clumio_phone_home" {
      + account_id               = "455372684561"
      + created_multi_region_cmk = true
      + id                       = (known after apply)
      + multi_region_cmk_key_id  = "mrk-c31e2af81d8c4f46a8989bd4c768afeb"
      + region                   = "us-west-2"
      + role_arn                 = (known after apply)
      + role_id                  = (known after apply)
      + token                    = "8454355f-edc6-4df7-afd9-31b2d73d6bc0"
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.clumio_byok.aws_iam_role.byok_mgmt_role: Creating...
module.clumio_byok.aws_iam_role.byok_mgmt_role: Creation complete after 1s [id=ClumioKMSRole-tf-Org7]
module.clumio_byok.clumio_post_process_kms.clumio_phone_home: Creating...
module.clumio_byok.clumio_post_process_kms.clumio_phone_home: Creation complete after 1s [id=455372684561/us-west-2/8454355f-edc6-4df7-afd9-31b2d73d6bc0]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
```

**Comment by deepak-clumio** [comment]
List wallets:
```
{
    "current_count": 0,
    "limit": 10,
    "start": "",
    "_links": {
        "_self": {
            "href": "/wallets",
            "type": "get",
            "templated": false
        },
        "_first": {
            "href": "/wallets?start=eyJTIjoiIiwiRiI6dHJ1ZX0%3D",
            "type": "get",
            "templated": false
        },
        "_last": {
            "href": "/wallets?start=eyJTIjoiIiwiRiI6ZmFsc2V9",
            "type": "get",
            "templated": false
        },
        "_prev": {
            "href": "/wallets?start=eyJTIjoiNDU1MzcyNjg0NTYxIiwiRiI6ZmFsc2V9",
            "type": "get",
            "templated": false
        }
    },
    "_embedded": {
        "items": [
            {
                "id": "455372684561",
                "account_native_id": "455372684561",
                "token": "8454355f-edc6-4df7-afd9-31b2d73d6bc0",
                "template_url": "https://clumio-be-uw-2-shared.s3.amazonaws.com/public/dep/backend-03-uw-2/cryptsrv/us-west-2/customer_kms_v2.yaml",
                "deployment_url": "https://us-west-2.console.aws.amazon.com/cloudformation/home?region=us-west-2#/stacks/create/review?param_ClumioToken=8454355f-edc6-4df7-afd9-31b2d73d6bc0&param_ExternalId=8be113c0-c9ff-41c7-98f3-a188e5c2ab99&param_RoleName=ClumioKMSRole-backend-03-uw-2-Org7&stackName=ClumioKMS-backend-03-uw-2-Org7&templateURL=https://clumio-be-uw-2-shared.s3.amazonaws.com/public/dep/backend-03-uw-2/cryptsrv/us-west-2/customer_kms_v2.yaml",
                "state": "Enabled",
                "role_arn": "arn:aws:iam::455372684561:role/ClumioKMSRole-tf-Org7",
                "error_code": "",
                "error_message": "",
                "stack_version": 0,
                "installed_regions": [
                    "us-west-2",
                    "us-east-1"
                ],
                "supported_regions": [
                    "us-west-2",
                    "us-east-1"
                ],
                "clumio_aws_account_id": "972749477403",
                "_links": {
                    "_self": {
                        "href": "/wallets/455372684561",
                        "type": "get",
                        "templated": false
                    }
                }
            }
        ]
    }
}
```

List keys:
```
{
    "current_count": 0,
    "limit": 10,
    "start": "",
    "_links": {
        "_self": {
            "href": "/wallets/455372684561/keys",
            "type": "get",
            "templated": false
        },
        "_first": {
            "href": "/wallets/455372684561/keys?start=eyJTIjoiIiwiRiI6dHJ1ZX0%3D",
            "type": "get",
            "templated": false
        },
        "_last": {
            "href": "/wallets/455372684561/keys?start=eyJTIjoiIiwiRiI6ZmFsc2V9",
            "type": "get",
            "templated": false
        },
        "_prev": {
            "href": "/wallets/455372684561/keys?start=eyJTIjoibXJrLWMzMWUyYWY4MWQ4YzRmNDZhODk4OWJkNGM3NjhhZmViIiwiRiI6ZmFsc2V9",
            "type": "get",
            "templated": false
        }
    },
    "_embedded": {
        "items": [
            {
                "id": "mrk-c31e2af81d8c4f46a8989bd4c768afeb",
                "cmk_id": "",
                "state": "Available",
                "multi_region_cmk_id": "mrk-c31e2af81d8c4f46a8989bd4c768afeb",
                "_links": {
                    "_self": {
                        "href": "/wallets/455372684561/keys/mrk-c31e2af81d8c4f46a8989bd4c768afeb",
                        "type": "get",
                        "templated": false
                    }
                }
            }
        ]
    }
}
```
```

File: versions.tf [hunk 1]
```diff
@@ -4,8 +4,7 @@ terraform {
       source  = "clumio-code/clumio"
       version = "~>0.3.0"
```
Comment (prakkarp) [inline versions.tf:5]: Change the version to ">=0.3.0" as we will soon be releasing version 0.4.0. If we have it as "~>0.3.0" it will only with 0.3.x.

**Review by prakkarp — COMMENTED** [review]
One comment. LGTM otherwise.

**Review by prakkarp — APPROVED** [review]

**Pull request merged**